### PR TITLE
[NFC] Use references to avoid copying

### PR DESCRIPTION
**Pull request by pratyay-p**
Modifying some uses of `auto` to `auto&` to avoid unnecessary copying.

### CODE REVIEW

**Comment by pratyay-p** [comment]
@KanRobert , @phoebewang , @MalaySanghi can you please review?

File: llvm/utils/TableGen/IntrinsicEmitter.cpp [hunk 1]
```diff
@@ -255,7 +255,7 @@ void IntrinsicEmitter::EmitIntrinsicToOverloadTable(
 static constexpr uint8_t OTable[] = {
   0
   )";
-  for (auto [I, Int] : enumerate(Ints)) {
+  for (const auto& [I, Int] : enumerate(Ints)) {
```
Comment (MalaySanghi) [inline llvm/utils/TableGen/IntrinsicEmitter.cpp:258]: enumerate already provides a reference.

Comment (pratyay-p) [reply llvm/utils/TableGen/IntrinsicEmitter.cpp]: I see, I have removed the reference in commit https://github.com/llvm/llvm-project/pull/110462/commits/a6e6e75ceba9ae37138a9364ecd4b38a22f30644.

File: llvm/utils/TableGen/IntrinsicEmitter.cpp [hunk 2]
```diff
@@ -229,8 +229,9 @@ struct IntrinsicTargetInfo {
 };
 static constexpr IntrinsicTargetInfo TargetInfos[] = {
 )";
-  for (const auto [Name, Offset, Count] : Ints.Targets)
-    OS << formatv("  {{\"{}\", {}, {}},\n", Name, Offset, Count);
+  for (const auto &Target : Ints.Targets)
```
Comment (phoebewang) [inline llvm/utils/TableGen/IntrinsicEmitter.cpp:232]: I recall references is not good for small value, maybe ignore them and just remove the `const`?

Comment (pratyay-p) [reply llvm/utils/TableGen/IntrinsicEmitter.cpp]: Done, removed reference and `const`.

File: llvm/utils/TableGen/IntrinsicEmitter.cpp [hunk 3]
```diff
@@ -256,7 +256,7 @@ void IntrinsicEmitter::EmitIntrinsicToOverloadTable(
 static constexpr uint8_t OTable[] = {
   0
   )";
-  for (auto [I, Int] : enumerate(Ints)) {
+  for (const auto [I, Int] : enumerate(Ints)) {
```
Comment (phoebewang) [inline llvm/utils/TableGen/IntrinsicEmitter.cpp:259]: Given it is just a copy, do we still need `const`?

Comment (pratyay-p) [reply llvm/utils/TableGen/IntrinsicEmitter.cpp]: Removed `const`

File: llvm/utils/TableGen/IntrinsicEmitter.cpp [hunk 4]
```diff
@@ -346,7 +346,7 @@ static constexpr {} IIT_Table[] = {{
                 FixedEncodingTypeName);
 
   unsigned MaxOffset = 0;
-  for (auto [Idx, FixedEncoding, Int] : enumerate(FixedEncodings, Ints)) {
+  for (const auto [Idx, FixedEncoding, Int] : enumerate(FixedEncodings, Ints)) {
```
Comment (phoebewang) [inline llvm/utils/TableGen/IntrinsicEmitter.cpp:349]: I don't know why leaving `const` here. I had a quick search. `const auto [` is much smaller used in LLVM code base.
```
$ grep -r 'const auto \[' llvm/| wc
     37     275    3760
$ grep -r 'auto \[' llvm/| wc
    817    5871   82133
```

Comment (pratyay-p) [reply llvm/utils/TableGen/IntrinsicEmitter.cpp:349]: Got it. It seems that we wont need to change anything in this PR. Closing this.

**Pull request closed without merging**